### PR TITLE
prom query via api

### PR DESCRIPTION
**Pull request by ajssmith**

### CODE REVIEW

**Comment by grs** [comment]
I may well be doing something wrong, but though the console comes up ok and doesn't give any errors, it doesn't show any metrics (I have a tcp exposed service over which I have made many http calls).

Also, oddly, I connected three sites, all on one cluster. Two were created using the code from this PR, one from master. The console only shows the two sites created using code from this PR. Though the cli status shows the other site connected, and it is routing connections as expected, it does not show up in the console.

As I say, perhaps I am doing something wrong in my testing here though.

**Comment by ajssmith** [comment]
With this change, an update to the console implementation is needed. Not sure that it would explain the behaviors you observed and will investigate.

File: cmd/flow-collector/handlers.go [hunk 1]
```diff
@@ -166,3 +168,45 @@ func (c *Controller) collectorHandler(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "%s", *response.Body)
 	}
 }
+
+func (c *Controller) promqueryHandler(w http.ResponseWriter, r *http.Request) {
+	client := http.Client{}
+
+	urlOut := c.FlowCollector.Collector.PrometheusUrl + "query?" + r.URL.RawQuery
+	proxyReq, err := http.NewRequest(r.Method, urlOut, nil)
```
Comment (fgiorgetti) [inline cmd/flow-collector/handlers.go:176]: Don't you need to pass the r.Body and r.Header to the proxyReq?
Without it, POST data won't be passed through or am I missing something?

Comment (ajssmith) [reply cmd/flow-collector/handlers.go:176]: Presently, the console does not POST to the query api

**Review by nluaces — APPROVED** [review]

**Comment by grs** [comment]
On the site not showing up, I tried with 1.4.0-rc3 and that site does show up ok. I'm also now seeing some metrics; not sure what happened to change that. Perhaps I was too impatient. Seems to be working ok now for me.

**Review by grs — APPROVED** [review]

**Pull request merged**